core/unix/os.c

```diff
@@ -2751,10 +2751,10 @@ static void
 replace_thread_id(thread_id_t old, thread_id_t new)
 {
 #ifdef HAVE_TLS
-    thread_id_t new_tid = new;
+    int32_t new_tid = (int32_t) new; /* can't use thread_id_t since it's 64-bits on x64 */
     ASSERT(is_thread_tls_initialized());
     DOCHECK(1, {
-        thread_id_t old_tid;
+        int32_t old_tid; /* can't use thread_id_t since it's 64-bits on x64 */
```

Comment:
But you'll read too much for Linux: pid_t is 32 bits even for x86_64.  Do you mean you'll read extra into a local and then truncate it (and assume little-endian)?